app/controllers/internal/AccountTypeController.scala

```diff
@@ -72,10 +75,20 @@ trait AccountTypeController extends FrontendController with Actions {
       case AffinityGroupValue.AGENT =>
         Future.successful(Ok(Json.toJson(AccountTypeResponse(AccountType.Agent))))
       case _ =>
-        ruleEngine.getLocation(ruleContext).value map { location =>
+        val engineResult: EngineResult = ruleEngine.getLocation(ruleContext)
+
+        val finalResult = engineResult.value map { location  =>
           val accountType = accountTypeBasedOnLocation(location)
           Ok(Json.toJson(AccountTypeResponse(accountType)))
         }
+
+        if (extendedLoggingEnabled) {
+          engineResult.run map {
+            case (auditInfo, _) => Logger.warn (s"[AIV-1264 (internal)] ${auditInfo.ruleApplied.getOrElse ("No rule applied.")}")
+          }
+        }
+
```

Comment:
I'm not particularly familiar with `WriterT`, but I think this will run the rules engine twice - it may be necessary to restructure it to only reference `engineResult` once

Maybe something like
```suggestion
        val futureLocation = engineResult.run.map {
          case (auditInfo, location) if extendedLoggingEnabled => Logger.warn(/*message*/); location
          case (_, location) => location
        }
        
        futureLocation.map { /* convert to result */ }
```

Reply:
I'm not sure the future will be run twice but it is a bit ugly to reference it twice. You could make use of the `run` method to get into the `AuditInfo` and the `Location` in one go - something like this:
```scala
ruleEngine.getLocation(ruleContext).run.map{ case (auditInfo, location) =>
          if (extendedLoggingEnabled) {
           Logger.warn (s"[AIV-1264 (internal)] ${auditInfo.ruleApplied.getOrElse ("No rule applied.")}")
          }

          val accountType = accountTypeBasedOnLocation(location)
          Ok(Json.toJson(AccountTypeResponse(accountType)))
        }
```
That's a cleaner way of implementing this.

Reply:
:+1: